sql/rpl_gtid_set.cc

```diff
@@ -304,6 +304,21 @@ void Gtid_set::clear_set_and_sid_map() {
   DBUG_VOID_RETURN;
 }
 
+void Gtid_set::remove(const rpl_sidno_container &sidnos) {
+  if (sid_lock != nullptr) sid_lock->assert_some_wrlock();
+  cached_string_length = -1;
```

Comment:
It would seem like `has_cached_string_length = false` should be the statement you need if you want to invalidate the string length?

Reply:
Fixed.